porter/main.py

```diff
@@ -171,16 +187,17 @@ def get_ursula_info(ursula_address) -> Porter.UrsulaInfo:
                 self.log.debug(f"Ursula ({ursula_address}) is unreachable: {str(e)}")
                 raise
 
-        self.block_until_number_of_known_nodes_is(quantity,
-                                                  timeout=self.execution_timeout,
-                                                  learn_on_this_thread=True,
-                                                  eager=True)
+        self.block_until_number_of_known_nodes_is(
+            quantity, timeout=timeout, learn_on_this_thread=True, eager=True
+        )
 
-        worker_pool = WorkerPool(worker=get_ursula_info,
-                                 value_factory=value_factory,
-                                 target_successes=quantity,
-                                 timeout=self.execution_timeout,
-                                 stagger_timeout=1)
+        worker_pool = WorkerPool(
+            worker=get_ursula_info,
+            value_factory=value_factory,
+            target_successes=quantity,
+            timeout=timeout,
+            stagger_timeout=1,
```

Comment:
What's this timeout?

Reply:
This is the timeout between obtaining batches of values for processing - https://github.com/nucypher/nucypher/blob/main/nucypher/utilities/concurrency.py#L322.

In this case batches of staking provider addresses.